src/Shared/NamedPipeUtil.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.IO;
+
+namespace Microsoft.Build.Shared
+{
+    internal static class NamedPipeUtil
+    {
+        internal static string GetPipeNameOrPath(string pipeName)
+        {
+            if (NativeMethodsShared.IsUnixLike)
+            {
+                // If we're on a Unix machine then named pipes are implemented using Unix Domain Sockets.
+                // Most Unix systems have a maximum path length limit for Unix Domain Sockets, with
+                // Mac having a particularly short one. Mac also has a generated temp directory that
+                // can be quite long, leaving very little room for the actual pipe name. Fortunately,
+                // '/tmp' is mandated by POSIX to always be a valid temp directory, so we can use that
+                // instead.
+                return Path.Combine("/tmp", pipeName);
```

Comment:
I remember fighting with this and I could have sworn we had some other fix. But I can't seem to find any evidence of that.

Reply:
Just clarifying, do I need to do anything here?

Reply:
Nope!